src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arrayStates = [];
+
+  for (const key of actions) {
```

Comment:
The variable name `key` is misleading in the context of iterating over `actions`. A more appropriate name would be `action` or `currentAction` to accurately represent the individual action being processed.